core/src/main/java/fr/sncf/osrd/cli/ReproduceRequest.kt

```diff
@@ -93,3 +106,22 @@ class ReproduceRequest : CliCommand {
         return 0
     }
 }
+
+/**
+ * Implement the InfraProvider interface using a railjson file. Used to reproduce requests without
+ * needing to run any other part of the stack.
+ */
+data class FileInfraProvider(val path: String) : InfraProvider {
+    override fun getInfra(
+        infraId: String?,
+        expectedVersion: String?,
+        diagnosticRecorder: DiagnosticRecorder?
+    ): FullInfra? {
```

Comment:
Nothing can return null here, and we've got a method for that :)
```suggestion
    ): FullInfra {
        val rjs = RJSParser.parseRailJSONFromFile(path)
        return FullInfra.fromRJSInfra(rjs, makeSignalingSimulator())
```

Reply:
Fixed

Reply:
I think you didn't see the second part of the fix with the return ;)

Reply:
Second part of the fix? I'm not sure I get it

Oh wait I just realized there's a `FullInfra.fromRJSInfra()`, I don't know if that's what you had in mind but I will use this. (done in last commit)